dev/tools/gen_defaults/lib/radio_template.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'template.dart';
+
+class RadioTemplate extends TokenTemplate {
+  const RadioTemplate(super.blockName, super.fileName, super.tokens, {
+    super.colorSchemePrefix = '_colors.',
+  });
+
+  @override
+  String generate() => '''
+class _RadioDefaultsM3 extends RadioThemeData {
+  _RadioDefaultsM3(BuildContext context)
+    : _theme = Theme.of(context),
+      _colors = Theme.of(context).colorScheme;
+
+  final ThemeData _theme;
+  final ColorScheme _colors;
+
+  @override
+  MaterialStateProperty<Color> get fillColor {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled)) {
+        return ${componentColor('md.comp.radio-button.disabled.unselected.icon')};
+      }
+      if (states.contains(MaterialState.selected)) {
+        return ${componentColor('md.comp.radio-button.selected.icon')};
+      }
+      if (states.contains(MaterialState.pressed)) {
+        return ${componentColor('md.comp.radio-button.unselected.pressed.icon')};
+      }
+      if (states.contains(MaterialState.hovered)) {
+        return ${componentColor('md.comp.radio-button.unselected.hover.icon')};
+      }
+      if (states.contains(MaterialState.focused)) {
+        return ${componentColor('md.comp.radio-button.unselected.focus.icon')};
+      }
+      return ${componentColor('md.comp.radio-button.unselected.icon')};
+    });
```

Comment:
Shouldn't we check for `MaterialState.selected` first as in `overlayColor` since there's 2 sets of tokens for unselected and selected?

Reply:
Yes, selected is handled before unselected here. The order here is firstly checking disabled states, then selected, then unselected in pressed/hovered/focused states. If the button is disabled, both selected and unselected states return a same value for the fillColor, so I only put a `unselected.icon` token in disabled condition.

Reply:
I meant to say, for example, that while we're handling the disabled unselected case, we're not handling the disabled case. Same goes for other selected states. `overlayColor` handles these states correctly.

Reply:
When the button is disabled, both selected and unselected tokens map to same value (color), so I only used the `md.comp.radio-button.disabled.unselected.icon` here for both selected and unselected cases. Should we handle each condition even if they have same values, or we can update them later only if the tokens become different?

Reply:
While we haven't been doing this in a disciplined way, if it is easy to support driving this from the tokens (even if the current tokens result in the same values), then we should probably do so.

Reply:
I see. Fixing it. Thanks!

Reply:
We should use tokens as if they were different, which they may be in future updates.

Reply:
Sure. Fixed it. Thanks!